File: app/src/main/java/me/instabattle/app/activities/MenuActivity.java
package me.instabattle.app.activities;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import me.instabattle.app.R;
import me.instabattle.app.services.LocationService;

public class MenuActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_menu);

        startService(new Intent(this, LocationService.class));
    }

    public void onButtonClick(View view) {
        Class<?> cls = null;
        switch(view.getId()) {
            case R.id.mapBtn:
                cls = MapActivity.class;
                MapActivity.gotHereFrom = MenuActivity.class;
                break;
            case R.id.battlesBtn:
                cls = BattleListActivity.class;
                break;
            case R.id.ratingBtn:
                cls = RatingActivity.class;
                break;
            case R.id.myProfileBtn:
                cls = MyProfileActivity.class;
                break;
        }
        Intent intent = new Intent(this, cls);
        startActivity(intent);
    }

    @Override
    public void onBackPressed() {
        //do nothing
    }
}

File: app/src/main/kotlin/me/instabattle/app/activities/MenuActivity.kt
package me.instabattle.app.activities

import android.os.Bundle
import android.view.View

import me.instabattle.app.R
import me.instabattle.app.services.LocationService
import org.jetbrains.anko.intentFor
import org.jetbrains.anko.startActivity

class MenuActivity : DefaultActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_menu)

        startService(intentFor<LocationService>())
    }

    fun onButtonClick(view: View) {
        when (view.id) {
            R.id.mapBtn -> {
                MapActivity.gotHereFrom = MenuActivity::class.java
                startActivity<MapActivity>()
            }
            R.id.battlesBtn -> startActivity<BattleListActivity>()
            R.id.myProfileBtn -> startActivity<MyProfileActivity>()
        }
    }

    override fun onBackPressed() {}
}
